packages/react/src/ConfirmationDialog/ConfirmationDialog.tsx

```diff
@@ -43,54 +46,85 @@ export interface ConfirmationDialogProps {
   confirmButtonType?: 'normal' | 'primary' | 'danger'
 }
 
-const StyledConfirmationHeader = styled.div`
-  padding: ${get('space.2')};
-  display: flex;
-  flex-direction: row;
-`
-const StyledTitle = styled(Box).attrs({as: 'h1'})`
-  font-size: ${get('fontSizes.3')};
-  font-weight: ${get('fontWeights.bold')};
-  padding: 6px ${get('space.2')};
-  flex-grow: 1;
-  margin: 0; /* override default margin */
-`
+const CSS_MODULES_FEATURE_FLAG = 'primer_react_css_modules_team'
+
+const StyledConfirmationHeader = toggleStyledComponent(
+  CSS_MODULES_FEATURE_FLAG,
+  'div',
+  styled.div`
+    padding: ${get('space.2')};
+    display: flex;
+    flex-direction: row;
+  `,
+)
+
+const StyledTitle = toggleStyledComponent(
+  CSS_MODULES_FEATURE_FLAG,
+  'h1',
+  styled(Box).attrs({as: 'h1'})`
+    font-size: ${get('fontSizes.3')};
+    font-weight: ${get('fontWeights.bold')};
+    padding: 6px ${get('space.2')};
+    flex-grow: 1;
+    margin: 0; /* override default margin */
+  `,
+)
+
 const ConfirmationHeader: React.FC<React.PropsWithChildren<DialogHeaderProps>> = ({title, onClose, dialogLabelId}) => {
   const onCloseClick = useCallback(() => {
     onClose('close-button')
   }, [onClose])
+
+  const enabled = useFeatureFlag(CSS_MODULES_FEATURE_FLAG)
   return (
-    <StyledConfirmationHeader>
+    <StyledConfirmationHeader className={enabled && classes.ConfirmationHeader}>
       <StyledTitle id={dialogLabelId}>{title}</StyledTitle>
       <Dialog.CloseButton onClose={onCloseClick} />
     </StyledConfirmationHeader>
   )
 }
-const StyledConfirmationBody = styled(Box)`
-  font-size: ${get('fontSizes.1')};
-  padding: 0 ${get('space.3')} ${get('space.3')} ${get('space.3')};
-  flex-grow: 1;
-`
+const StyledConfirmationBody = toggleStyledComponent(
+  CSS_MODULES_FEATURE_FLAG,
+  'div',
+  styled(Box)`
+    font-size: ${get('fontSizes.1')};
+    padding: 0 ${get('space.3')} ${get('space.3')} ${get('space.3')};
+    flex-grow: 1;
+  `,
+)
+
 const ConfirmationBody: React.FC<React.PropsWithChildren<DialogProps>> = ({children}) => {
-  return <StyledConfirmationBody>{children}</StyledConfirmationBody>
+  const enabled = useFeatureFlag(CSS_MODULES_FEATURE_FLAG)
+  return <StyledConfirmationBody className={enabled && classes.ConfirmationBody}>{children}</StyledConfirmationBody>
```

Comment:
Since I don't believe we allow custom styling for this component, I didn't bother adding a `className` prop. Not sure if we should always add one regardless or not if the component doesn't utilize `sx` or custom styles?